funnel/templates/siteadmin_comments.html.jinja2

```diff
@@ -65,6 +79,16 @@
     {{ rendersubmit([("spam-comments-bulk", "Mark spam", '')]) }}
   </div>
 </form>
+
+<div class="mui--text-center">
+  {% for page in pages[:10] %}
+    <a
+      class="btn-pagination-item mui-btn {% if page != current_page %}mui-btn--accent{% endif %}"
+      href="{{ url_for('siteadmin_comments') }}?query={{ query }}&page={{ page }}"
+      >{{ page }}</a>
```

Comment:
Queries have built-in batch support. See the search views for example of how to use.